### DISCUSSION TITLE
Unable to use the Jekyll {seo} tag with Pages

### DISCUSSION THREAD

**Discussion by lukeylias** [community/community#108451] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi, I've been trying to add SEO to my published website on GitHub Pages. I've been troubleshooting for days, any assistance would be much appreciated.

I've followed the following steps:

1. Created _config.yml file and added `Plugins:   - jekyll-seo-tag`
2. Created a Gemfile with the following source 'https://rubygems.org'
gem 'jekyll', '~> 4.2.0' # Specify the Jekyll version you're using
gem 'jekyll-seo-tag'   
3. Ran  `bundle install --path vendor/bundle`
4. Added {seo} into my `<header>`
5. Added Front matter

When I preview the site I can visibly see the `------ {seo} tag`
![CleanShot 2024-02-16 at 13 17 40@2x](https://github.com/community/community/assets/58713241/89e0983e-c5b9-4465-bf86-bc078a4dab01)

I also am getting `<meta charset="UTF-8" /> meta element should be specified in <header> and not <body> – error `

Heres my code 

```
---
---

<!DOCTYPE html>
<html lang="en">
  <head>
    <title>Sa11y: Simple Accessibility Guidance for Beginners</title>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <meta
      name="description"
      content="Sa11y is a tool that provides easy-to-follow accessibility guidance with step-by-step instructions tailored for beginners to effectively tackle accessibility issues."
    />

    <link rel="stylesheet" href="styles.css" />
    <link
      rel="stylesheet"
      href="https://fonts.googleapis.com/css2?family=Lato:wght@400;700&display=swap"
    />
    <link
      href="https://fonts.googleapis.com/icon?family=Material+Icons"
      rel="stylesheet"
    />
    <link
      rel="stylesheet"
      href="https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined:opsz,wght,FILL,GRAD@20..48,100..700,0..1,-50..200"
    />
    <link rel="icon" href="assets/sa11y-favicon.svg" type="image/x-icon" />
    <link
      href="https://cdn.jsdelivr.net/npm/bootstrap-icons@1.7.2/font/bootstrap-icons.css"
      rel="stylesheet"
    />
    {% seo %}
  </head>
```

**Comment by yoannchaudet** [comment]
How do you deploy your Page site?

If you just selected a source branch for instance, you don't even need to install anything because we load the SEO plugin for you already: https://pages.github.com/versions/.

**Reply by lukeylias** [reply]
I just published it from the main branch. Am I meant to use GitHub actions?
![CleanShot 2024-02-17 at 08 45 09@2x](https://github.com/community/community/assets/58713241/47487864-3464-4d31-a550-9dc34bca6bed)

**Reply by yoannchaudet** [reply]
Gotcha so you don't exactly need all the plugins stuff, it's there for you already (not that it bothers to have it defined).

I think Jekyll does not expand Liquid syntax in an html file if there is no [front matter](https://jekyllrb.com/docs/front-matter/) for it.

https://jekyllrb.com/tutorials/orderofinterpretation/#order-of-interpretations

So at the top of your file (https://github.com/lukeylias/sa11y/blob/56f5f8c0994e862f78189fdaed8a30081c694310/index.html), I would try to add an empty front matter (like you have in your snippet here 🙈):

```
---
---
```

**Reply by lukeylias** [reply]
Thanks for the response, when I add the front matter, I get the following error messages
![CleanShot 2024-02-17 at 09 22 51@2x](https://github.com/community/community/assets/58713241/10a99956-1b4e-4139-a06d-51de5884aed0)

I'll push my latest changes into my repo now. [https://github.com/lukeylias/sa11y](url)

**Reply by yoannchaudet** [reply]
Your text editor does not understand what's a Jekyll front matter but that's valid for Jekyll.

As a matter of fact, your site looks great now ;) https://sa11y.com

I see all the SEO tags now!

<img width="934" alt="image" src="https://github.com/community/community/assets/14911070/4bf0bd8b-6866-4dbc-9730-68e9261ef07c">

**Comment by lukeylias** [comment]
It's really odd, I also can still see the tag and front matter when viewing the site, it's putting it into the body 😭
![CleanShot 2024-02-17 at 09 27 03@2x](https://github.com/community/community/assets/58713241/b5f80657-9297-4864-b9b5-37c6f584d5a9)

**Reply by yoannchaudet** [reply]
It's cached. Just hit `ctrl (or command on a mac) + R`.

**Reply by lukeylias** [reply]
OMG IT WORKS!! YES!!

**Reply by lukeylias** [reply]
Thank you so much, this is my first published site so I'm very new to it all. Really appreciate the responses <3
